tests/vmi_hook_sidecar_test.go

```diff
@@ -258,3 +286,27 @@ func RenderInvalidSMBiosSidecar() map[string]string {
 		"smbios.vm.kubevirt.io/baseBoardManufacturer": "Radical Edward",
 	}
 }
+
+func RenderSidecarWithConfigMap(version string) map[string]string {
+	return map[string]string{
+		"hooks.kubevirt.io/hookSidecars": fmt.Sprintf(`[{"args": ["--version", "%s"], "image":"%s/%s:%s", "configMap": {"name": "%s","key": "%s", "hookPath": "/usr/bin/onDefineDomain"}}]`,
+			version, flags.KubeVirtUtilityRepoPrefix, sidecarShimImage, sidecarShimImageTag, configMapName, configMapKey),
```

Comment:
> How can I use [quay.io/kubevirt/sidecar-shim](http://quay.io/kubevirt/sidecar-shim) image in test? It doesn't have a latest tag, but using the tag 20231026_f3f020dd2 the test works just fine.

In tests, you generally do not use images from quay.io. E.g. `sidecar-shim` is part of KubeVirt sources, hence it is built during a test job and then pushed to a local registry. A hardcoded tag will not work in such case (and in environments which are not using quay.io). The image tag needs to be `flags.KubeVirtUtilityVersionTag` so it can be configured via cmdline args:

```suggestion
			version, flags.KubeVirtUtilityRepoPrefix, sidecarShimImage, flags.KubeVirtUtilityVersionTag, configMapName, configMapKey),
```

Ref: https://github.com/kubevirt/kubevirt/blob/d7371feb56fcb138806b1b6e5b6ab154e900a2e0/tests/vmi_hook_sidecar_test.go#L250

This should also work when you do local testing with `make cluster-sync && make functests`. In that case, the images will be pushed to a local registry respecting DOCKER_PREFIX and DOCKER_TAG env vars.

Reply:
> In tests, you generally do not use images from quay.io. E.g. `sidecar-shim` is part of KubeVirt sources, hence it is built during a test job and then pushed to a local registry. A hardcoded tag will not work in such case (and in environments which are not using quay.io).

This is how I expected it to be. Thanks for explaining.

> This should also work when you do local testing with `make cluster-sync && make functests`. In that case, the images will be pushed to a local registry respecting DOCKER_PREFIX and DOCKER_TAG env vars.

I can't make the test I have proposed in this PR to pass locally after making the change you suggested. Maybe I'm doing something wrong. It's timing out while starting the VMI. But I have pushed the changes nonetheless, since I might be doing something wrong. What I tried is:
```
$ DOCKER_PREFIX=registry:5000 make cluster-sync functest
```

Reply:
I had made an [incorrect change](https://github.com/kubevirt/kubevirt/commit/13f3a43b7f5132d3b82f908ddeacc6d908facf01) and hence couldn't get the tests to pass. 🤦🏽‍♂️

Reply:
> BTW, you can also interact with the cluster while running the tests, so you can easily debug failures (e.g. get logs). Just run:
> 
> ```
> export KUBECONFIG=$(./cluster-up/kubeconfig.sh)
> ```
> 
> And then your kubectl will be directed to the test cluster.

If you don't mind entertaining a possibly naive question, I was initially running tests through GoLand IDE and not via `make functest` (so that I could step through).

I have exported `-kubeconfig` (to be `_ci-configs/k8s-1.27/.kubeconfig`) and `-config` (to be `tests/default-config.json`) arguments in the configuration. The `test_id:3157` test still fails when running through GoLand, but passes just fine with `make functest`. I tried setting as well as removing values for `DOCKER_PREFIX` (to be `registry:5000`) and `DOCKER_TAG` (to be `devel`), but neither helps.

I feel like I'm missing something silly here. 😕